### PR TITLE
OSDOCS#7770: Migrate API Data Protection for ROSA content from MOBB to ROSA docs

### PR DESCRIPTION
**Pull request by skopacz1**
Version(s): 4.13+

Issue: [OSDOCS-7770](https://issues.redhat.com/browse/OSDOCS-7770)

This PR migrates content for "Deploying OpenShift API for Data Protection on a ROSA cluster" from the MOBB ninja docs to the ROSA docs

QE review:
- [ ] QE has approved this change.

Preview: [Deploying OpenShift API for Data Protection on a ROSA cluster](https://64864--docspreview.netlify.app/openshift-rosa/latest/rosa_tutorials/rosa-mobb-deploy-api-data-protection)

### CODE REVIEW

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc
Comment (skopacz1) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:1]: I am sure that for many of the steps with larger chunks of commands to paste into the CLI, I missed discrete commands that should have their own "$" at the beginning of the line. Would appreciate help identifying which lines need to have that character added.

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+       config:
+         credentialsFile: /tmp/credentials/openshift-adp/cloud-credentials-credentials
+         enableSharedConfig: 'true'
+         profile: default
+         region: ${REGION}
+       provider: aws
+EOF
+----
+
+[NOTE]
```
Comment (skopacz1) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:341]: Is this note for the entire procedure or for the step that it's under?

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+
+. Remove the backup and restore resources from the cluster if they are no longer required:
++
+[source,terminal]
+----
+$ oc delete backup hello-world
+$ oc delete restore hello-world
+----
+
+. To delete the backup/restore and remote objects in s3:
```
Comment (skopacz1) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:553]: Is this supposed to be a separate step or a variation of the step above?

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+//authors:
+//  - Paul Czarkowski
+//  - Dustin Scott
+//---
+
+include::snippets/mobb-support-statement.adoc[leveloffset=+1]
+
+.Prerequisites
+
+* A ROSA classic cluster
```
Comment (bmcelvee) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:24]: I think we can link out to the product docs until we bring this over (optional)
 
E.g., https://docs.openshift.com/rosa/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.html

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 4]
```diff
@@ -0,0 +1,591 @@
...
+// The prereq above is originally linked to a mobb doc (which in turn also points to an OCP doc about the same topic). Not clear whether this prereq should be linked to either of those pages in this PR.
+
+.Environment
+
+* Prepare the environment variables:
++
+[NOTE]
+====
+Change the cluster name to match your ROSA cluster and ensure you're logged into the cluster as an Administrator.
```
Comment (bmcelvee) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:33]: s/you're/you are 

Just formalizing the language a tiny bit.

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 5]
```diff
@@ -0,0 +1,591 @@
...
+
+== Prepare AWS Account
+
+. Create an IAM Policy to allow for S3 Access:
++
+[source,terminal]
+----
+$ POLICY_ARN=$(aws iam list-policies --query "Policies[?PolicyName=='RosaOadpVer1'].{ARN:Arn}" --output text)
+if [[ -z "${POLICY_ARN}" ]]; then
+cat << EOF > ${SCRATCH}/policy.json
```
Comment (bmcelvee) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:60]: Add a $ before `cat` - it's a sneaky one

$ cat << EOF > ${SCRATCH}/policy.json

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 6]
```diff
@@ -0,0 +1,591 @@
...
+     "ec2:CreateTags",
+     "ec2:CreateVolume",
+     "ec2:CreateSnapshot",
+     "ec2:DeleteSnapshot"
+   ],
+   "Resource": "*"
+ }
+]}
+EOF
+POLICY_ARN=$(aws iam create-policy --policy-name "RosaOadpVer1" \
```
Comment (bmcelvee) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:97]: Another sneaky one:

$ POLICY_ARN=$(aws iam create-policy --policy-name "RosaOadpVer1" \

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 7]
```diff
@@ -0,0 +1,591 @@
...
+      "StringEquals": {
+         "${OIDC_ENDPOINT}:sub": [
+           "system:serviceaccount:openshift-adp:openshift-adp-controller-manager",
+           "system:serviceaccount:openshift-adp:velero"]
+      }
+    }
+  }]
+}
+EOF
+ROLE_ARN=$(aws iam create-role --role-name \
```
Comment (bmcelvee) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:128]: $ ROLE_ARN=$(aws iam create-role --role-name \

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 8]
```diff
@@ -0,0 +1,591 @@
...
+ "${ROLE_NAME}" \
+  --assume-role-policy-document file://${SCRATCH}/trust-policy.json \
+  --tags Key=rosa_cluster_id,Value=${ROSA_CLUSTER_ID} Key=rosa_openshift_version,Value=${CLUSTER_VERSION} Key=rosa_role_prefix,Value=ManagedOpenShift Key=operator_namespace,Value=openshift-adp Key=operator_name,Value=openshift-oadp \
+  --query Role.Arn --output text)
+
+echo ${ROLE_ARN}
```
Comment (bmcelvee) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:134]: $ echo ${ROLE_ARN}

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 9]
```diff
@@ -0,0 +1,591 @@
...
+web_identity_token_file = /var/run/secrets/openshift/serviceaccount/token
+EOF
+$ oc -n openshift-adp create secret generic cloud-credentials \
+ --from-file=${SCRATCH}/credentials
+----
+
+. Deploy the OADP Operator:
++
+[NOTE]
+====
```
Comment (bmcelvee) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:170]: Optional: I think there are some hard returns in the note that can be removed. No big deal either way :)

Comment (bmcelvee) [reply rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:170]: Optional: Double space "status.  This does not seem"

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 10]
```diff
@@ -0,0 +1,591 @@
...
+----
+$ oc get storageclass
+NAME                PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
+gp2                 kubernetes.io/aws-ebs   Delete          WaitForFirstConsumer   true                   4d21h
+gp2-csi             ebs.csi.aws.com         Delete          WaitForFirstConsumer   true                   4d21h
+gp3                 ebs.csi.aws.com         Delete          WaitForFirstConsumer   true                   4d21h
+gp3-csi (default)   ebs.csi.aws.com         Delete          WaitForFirstConsumer   true                   4d21h
+----
++
+Using either gp3-csi, gp2-csi, gp3 or gp2 will work.
```
Comment (bmcelvee) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:258]: s/we recommend including/include

**Comment by bmcelvee** [comment]
A few suggestions, but overall LGTM!

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 11]
```diff
@@ -0,0 +1,592 @@
...
++
+[NOTE]
+====
+Change the cluster name to match your ROSA cluster and ensure you are logged into the cluster as an Administrator.
+Ensure all fields are outputted correctly before moving on.
+====
++
+[source,terminal]
+----
+$ export CLUSTER_NAME=my-cluster
```
Comment (michaelryanmcneill) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:38]: replace with to make it easier for the user:
```
$ export CLUSTER_NAME=$(oc get infrastructure cluster -o=jsonpath="{.status.infrastructureName}"  | sed 's/-[a-z0-9]\{5\}$//')
```

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 12]
```diff
@@ -0,0 +1,592 @@
...
+----
+$ oc -n openshift-adp patch cloudstorage ${CLUSTER_NAME}-oadp -p '{"metadata":{"finalizers":null}}' --type=merge
+----
+====
+
+. Remove the Operator if it is no longer required:
++
+[source,terminal]
+----
+$ oc -n openshift-adp delete subscription oadp-operator
```
Comment (michaelryanmcneill) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:537]: replace with:

```
$ oc -n openshift-adp delete subscription redhat-oadp-operator
```

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc (hunk 2 above)
Comment (michaelryanmcneill) [reply rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:555]: This deletes the actual resources in S3 as opposed to just deleting the CR.

File: rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc [hunk 13]
```diff
@@ -0,0 +1,592 @@
...
+$ oc delete restore hello-world
+----
+
+. To delete the backup/restore and remote objects in s3:
++
+[source,terminal]
+----
+$ velero backup delete hello-world
+$ velero restore delete hello-world
+----
```
Comment (michaelryanmcneill) [inline rosa_tutorials/rosa-mobb-deploy-api-data-protection.adoc:561]: Move this section above the "Delete the Data Protection Application" step above.

**Review by michaelryanmcneill — CHANGES_REQUESTED** [review]
A few minor changes needed before publishing, thanks @skopacz1!

**Pull request merged**